### PR TITLE
Fix: Use alternative syntax in view scripts

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] adds a few semicolons (for consistency)
* [x] uses alternative syntax in view scripts (also for consistency)
* [x] removes an assignment and a condition with effect, but no use
* [x] while at it, escapes a few unescaped strings in the view templates affected above

### CODE REVIEW

File: module/User/view/scn-social-auth/user/login.phtml [hunk 1]
```diff
@@ -3,13 +3,11 @@
...
-        if ($socialSignIn) {
-            $socialSignIn = false;
-        }
-        echo '<p>' . $this->socialSignInButton($provider) . '</p>';
-    }
-    ?>
+    <?php $socialSignIn = true; ?>
+    <?php foreach ($this->options->getEnabledProviders() as $provider): ?>
+        <?php if ($socialSignIn): ?>
+            <?php $socialSignIn = false; ?>
```
Comment (ins0) [inline module/User/view/scn-social-auth/user/login.phtml]: someone know for this var is used?

Comment (localheinz) [inline module/User/view/scn-social-auth/user/login.phtml]: Maybe some

magic

?

Removing it in a second!

File: module/User/view/zfc-user/user/index.phtml [hunk 2]
```diff
@@ -22,14 +22,11 @@
...
-                    <div class="alert alert-block">No modules was submitted yet</div>
-                    <?php
-                }
-                foreach ($modules as $module) {
-                    echo $this->moduleView([
+                <?php if (empty($modules)): ?>
+                <div class="alert alert-block">No modules was submitted yet</div>
+                <?php endif; ?>
+                <?php foreach ($modules as $module): ?>
+                <?php echo $this->moduleView([
```
Comment (ins0) [inline module/User/view/zfc-user/user/index.phtml]: surrounded with an if else ``if (count($modules)...){ else ...`` we can skip the foreach and is cleaner

File: module/User/view/zfc-user/user/index.phtml [hunk 3]
```diff
@@ -37,22 +34,22 @@
...
-                <?php echo $this->userOrganizations() ?>
+                <?php echo $this->userOrganizations(); ?>
             </div>
         </div>
     </div>
     <div class="col-md-4">
         <div class="sidebar" style="text-align: center">
-            <h3 style="text-align:center">Hello, <?php echo $this->zfcUserDisplayName() ?>!</h3>
-            <img class="thumbnail" src="<?php echo $this->zfcUserIdentity()->getPhotoUrl() ?>" alt="<?php echo $this->zfcUserDisplayName() ?>"/>
+            <h3 style="text-align:center">Hello, <?php echo $this->zfcUserDisplayName(); ?>!</h3>
```
Comment (ins0) [inline module/User/view/zfc-user/user/index.phtml]: don't know but is this output safe or need escaped?

File: module/User/view/zfc-user/user/index.phtml [hunk 4]
```diff
@@ -37,22 +34,22 @@
...
+            <img class="thumbnail" src="<?php echo $this->zfcUserIdentity()->getPhotoUrl(); ?>" alt="<?php echo $this->zfcUserDisplayName(); ?>"/>
```
Comment (ins0) [inline module/User/view/zfc-user/user/index.phtml]: see above

File: module/User/view/scn-social-auth/user/login.phtml (hunk 1 above)
Comment (localheinz) [inline module/User/view/scn-social-auth/user/login.phtml]: Looks like this was mostly copied from

```php
<?php
$socialSignIn = true;
foreach ($this->options->getEnabledProviders() as $provider) {
    if ($socialSignIn) {
        echo '<h1>Social Sign In</h1>';
        $socialSignIn = false;
    }
    echo '<dd>' . $this->socialSignInButton($provider, $this->redirect) . '</dd>';
}
if ($this->options->getSocialLoginOnly() == false) {
    echo $this->zfcUserLogin;
}
```

See https://github.com/SocalNick/ScnSocialAuth/blob/master/view/scn-social-auth/user/login.phtml#L1-L12.

Comment (ins0) [inline module/User/view/scn-social-auth/user/login.phtml]: looks like its safe to

bye

File: module/User/view/zfc-user/user/index.phtml (hunk 3 above)
Comment (localheinz) [inline module/User/view/zfc-user/user/index.phtml]: I'd escape it, but probably out of the scope of this PR. However, will fix it right away! :wink:

**Comment by localheinz** [comment]
@ins0 

Thank you for reviewing, implemented the suggested changes!

**Comment by Ocramius** [comment]
This requires a rebase

**Comment by localheinz** [comment]
@Ocramius 

Rebased!

Thanks a lot for reviewing everything!

File: module/User/view/zfc-user/user/index.phtml [hunk 5]
```diff
@@ -22,37 +22,34 @@
         <br />
         <div class="tab-content">
             <div class="tab-pane active" id="modules">
-                <?php
-                if (empty($modules)) {
+            <?php if (0 === count($modules)): ?>
+                <div class="alert alert-block">No modules have been submitted yet</div>
+            <?php else: ?>
+                <?php foreach ($modules as $module): ?>
+                    <?php echo $this->moduleView([
```
Comment (Ocramius) [inline module/User/view/zfc-user/user/index.phtml]: Why does this helper require so many parameters? Isn't it easier to pass in `$module` instead?

File: module/User/view/zfc-user/user/index.phtml [hunk 6]
```diff
@@ -60,12 +57,4 @@
...
-
-<?php
-$url = $this->url('zf-module');
-$this->inlineScript()->appendScript(<<<EOT
-
-$("#repositories").load("$url");
-
-EOT
-);
+<?php $this->inlineScript()->appendScript('$("#repositories").load("' . $this->url('zf-module') . '");'); ?>
```
Comment (Ocramius) [inline module/User/view/zfc-user/user/index.phtml]: Should use `$this->escapeJs()` here

File: module/User/view/zfc-user/user/index.phtml [hunk 7]
```diff
@@ -11,7 +11,7 @@
             </div>
         </div>
         <?php foreach ($this->flashMessenger()->getMessages() as $message): ?>
-            <h3 class="zf-green"><?php echo $message ?></h3>
+            <h3 class="zf-green"><?php echo $message; ?></h3>
```
Comment (Ocramius) [inline module/User/view/zfc-user/user/index.phtml]: Should `$this->escapeHtml()` probably, or use the flash messenger directly.

File: module/User/view/zfc-user/user/index.phtml (hunk 5 above)
Comment (localheinz) [inline module/User/view/zfc-user/user/index.phtml]: @Ocramius 

That's what I plan to resolve next! :wink:

File: module/User/view/zfc-user/user/index.phtml (hunk 6 above)
Comment (localheinz) [inline module/User/view/zfc-user/user/index.phtml]: @Ocramius 

Will fix tomorrow morning! Have a good night!

File: module/User/view/zfc-user/user/index.phtml [hunk 8]
```diff
@@ -11,7 +11,7 @@
             </div>
         </div>
         <?php foreach ($this->flashMessenger()->getMessages() as $message): ?>
-            <h3 class="zf-green"><?php echo $message ?></h3>
+            <h3 class="zf-green"><?php echo $this->escapeHtml($message); ?></h3>
```
Comment (localheinz) [inline module/User/view/zfc-user/user/index.phtml]: @Ocramius 

Escaping for now, would like to re-iterate on this and use the `FlashMessenger` directly, as suggested! Also, I would like to use http://getbootstrap.com/components/#alerts, what do you think?

Comment (Ocramius) [inline module/User/view/zfc-user/user/index.phtml]: Can be done in a different MR then (with screenshots 'n stuff, so someone with better taste than me can deal with it :-) )

Comment (ins0) [inline module/User/view/zfc-user/user/index.phtml]: would like to do this if @localheinz agrees - as i allready implemented this on a diff project (personal) :smirk_cat: :

Comment (localheinz) [inline module/User/view/zfc-user/user/index.phtml]: @ins0 

Sure, go ahead!

Comment (localheinz) [inline module/User/view/zfc-user/user/index.phtml]: @ins0 

If you do, please have a look at places where classes are missing:

```
<div class="alert alert-block">No modules found</div>
```

which results in 

screen shot 2015-02-17 at 16 16 38

vs.

```
<div class="alert alert-info alert-block">No modules found</div>
```

which would result in 
 
screen shot 2015-02-17 at 16 16 56

:bulb: Just as an idea, in [**Getting Real**](https://basecamp.com/books/Getting%20Real.pdf) from 37signals, there's a chapter *The Blank State* - I think we could improve on that.

**Pull request merged**